kedro-datasets/kedro_datasets/matplotlib/matplotlib_writer.py

```diff
@@ -17,14 +18,13 @@
     get_filepath_str,
     get_protocol_and_path,
 )
+from matplotlib.figure import Figure
 
 from kedro_datasets._typing import ImagePreview
 
 
 class MatplotlibWriter(
-    AbstractVersionedDataset[
-        Union[plt.figure, list[plt.figure], dict[str, plt.figure]], NoReturn
-    ]
+    AbstractVersionedDataset[Union[Figure, list[Figure], dict[str, Figure]], NoReturn]
```

Comment:
Is it intentional to keep `Union` here?

Reply:
Updated!

Reply:
Turns out it is needed here, the python <=3.9 tests fail otherwise